applications/CoSimulationApplication/python_scripts/data_transfer_operators/3d_1d_data_transfer.py

```diff
@@ -0,0 +1,139 @@
+            model_part_origin      = from_solver_data.GetModelPart()
+            model_part_destination = to_solver_data.GetModelPart()
+
+            if self.echo_level > 0:
+                info_msg  = "Creating Mapper:\n"
+                info_msg += '    Origin: ModelPart "{}" of solver "{}"\n'.format(model_part_origin_name, from_solver_data.solver_name)
+                info_msg += '    Destination: ModelPart "{}" of solver "{}"'.format(model_part_destination_name, to_solver_data.solver_name)
+
+                cs_tools.cs_print_info(colors.bold(self._ClassName()), info_msg)
+
+            # Creating parameters for the data transfer
+            # TODO: I should check if the parameters change between different data transfers
+            parameters = KM.Parameters(self.settings["3d_1d_data_transfer_settings"].WriteJsonString())
+            if not parameters.Has("origin_variables"):
+                parameters.AddEmptyArray("origin_variables")
+            parameters["origin_variables"].Append(variable_origin.Name())
+            if not parameters.Has("destination_variables"):
+                parameters.AddEmptyArray("destination_variables")
+            parameters["destination_variables"].Append(variable_destination.Name())
+            for transfer_option in transfer_options.GetStringArray():
+                if transfer_option == "swap_sign":
+                    parameters["swap_sign"].SetBool(True)
+
+            # Check if the origin model part is 3D or 1D
+            if self.__check_model_part_3D(model_part_origin):
+                self.origin_is_3d = True
+            else:
+                self.origin_is_3d = False
+            # Clone is necessary because the settings are validated and defaults assigned, which could influence the creation of other data transfers
+            self.__data_transfer_process[identifier_tuple] = KratosCoSim.DataTransfer3D1DProcess(model_part_origin, model_part_destination, parameters.Clone())
+            self.__data_transfer_process[inverse_identifier_tuple] = KratosCoSim.DataTransfer3D1DProcess(model_part_origin, model_part_destination, parameters.Clone())
+
+            # Execute the data transfer
+            if self.origin_is_3d:
+                self.__data_transfer_process[identifier_tuple].Set(KM.Mapper.USE_TRANSPOSE)
+            self.__data_transfer_process[identifier_tuple].Execute()
+            if self.origin_is_3d:
+                self.__data_transfer_process[identifier_tuple].Reset(KM.Mapper.USE_TRANSPOSE)
+
+    def _Check(self, from_solver_data, to_solver_data):
+        def CheckData(data_to_check):
+            if "node" not in data_to_check.location:
+                raise Exception('Transfer only supports nodal values!"{}"\nChecking ModelPart "{}" of solver "{}"'.format(self._ClassName(), data_to_check.model_part_name, data_to_check.solver_name))
+
+        CheckData(from_solver_data)
+        CheckData(to_solver_data)
+
+    def __check_model_part_3D(self, model_part):
```

Comment:
what if the ModelPart has 1D and 3D elements?

Reply:
I will raise an exception